pkg/provisioner/ironic/ironic.go

```diff
@@ -1455,7 +1455,6 @@ func (p *ironicProvisioner) Provision(hostConf provisioner.HostConfigData) (resu
 			"metal3-namespace": p.host.ObjectMeta.Namespace,
 			"metal3-name":      p.host.ObjectMeta.Name,
 			"local-hostname":   p.host.ObjectMeta.Name,
-			"local_hostname":   p.host.ObjectMeta.Name,
```

Comment:
I would maybe not remove that. There was an issue quite some time ago, where one of the two was used by cloud-init instead of the other. I don't remember the specifics

Reply:
Although it passed all the 4 CI runs... I take you suggestion and keep it as it is.

Reply:
When this line was not here, tests were passing, now I am seeing this after pivot back : 

```
{"level":"info","ts":1617200333.4666467,"logger":"controllers.BareMetalHost","msg":"registering and validating access to management controller","baremetalhost":"metal3/node-1","provisioningState":"provisioned","credentials":{"credentials":{"name":"node-1-bmc-secret","namespace":"metal3"},"credentialsVersion":"8800"}}
{"level":"info","ts":1617200333.5034752,"logger":"provisioner.ironic","msg":"node with name metal3-node-1 doesn't exist","host":"metal3-node-1"}
{"level":"info","ts":1617200333.5035117,"logger":"provisioner.ironic","msg":"looking for existing node by MAC","host":"metal3-node-1","MAC":"00:b0:5e:58:de:6b"} 

baremetal node list
| c222e16a-0211-4806-8820-a4a635f47270 | node-1        | b5eea54f-078a-4cf9-8040-34f7638cfbad | power on    | manageable         | False       |
| 0acc9a34-feff-4286-af3d-60fa9dab14ad | metal3-node-0 | bc7538b7-6b86-4c0a-8c4d-903ac73adfc3 | power on    | active             | False 
```